examples/run_IR.py

```diff
@@ -50,56 +70,78 @@
 )
 # }}}
 # {{{run IR
-vd_data = None
-for vd_idx, vd in enumerate(vd_list_us):
-    vd_data = run_IR(
-        nScans=config_dict["nScans"],
-        indirect_idx=vd_idx,
-        indirect_len=len(vd_list_us),
-        ph1_cyc=ph1,
-        ph2_cyc=ph2,
-        adcOffset=config_dict["adc_offset"],
-        carrierFreq_MHz=config_dict["carrierFreq_MHz"],
-        nPoints=nPoints,
-        nEchoes=config_dict["nEchoes"],
-        vd=vd,
-        p90_us=config_dict["p90_us"],
-        repetition_us=config_dict["repetition_us"],
-        tau_us=config_dict["tau_us"],
-        SW_kHz=config_dict["SW_kHz"],
-        ret_data=vd_data,
+with power_control() as p:
+    dip_f = p.dip_lock(
+        config_dict["uw_dip_center_GHz"] - config_dict["uw_dip_width_GHz"] / 2,
+        config_dict["uw_dip_center_GHz"] + config_dict["uw_dip_width_GHz"] / 2,
+    )
+    dip_f /= 1e9
+    p.set_power(dB_setting)
+    for k in range(10):
+        time.sleep(0.5)
+        if p.get_power_setting() >= dB_setting:
+            break
+    if p.get_power_setting() < dB_setting:
+        raise ValueError("After 10 tries, this power has still not settled")
+    time.sleep(5)
+    meter_power = p.get_power_setting()
+    vd_data = None
+    for vd_idx, vd in enumerate(vd_list_us):
+        vd_data = run_IR(
+            nScans=config_dict["nScans"],
```

Comment:
can now run IR at any specified power instead rather than not interacting with the B12 at all